user_manager/api/v1/views.py

```diff
@@ -0,0 +1,419 @@
+
+    Args:
+        queryset(QuerySet): UserManagerRole queryset
+        user_id(str): username or email address of user
+    Returns:
+        queryset filtered by user
+    """
+
+    if user_id is None:
+        return queryset
+    elif '@' in user_id:
+        return queryset.filter(user__email=user_id)
+    else:
+        return queryset.filter(user__username=user_id)
+
+
+class ManagerViewMixin(object):
+    """
+    Provide common functionality for all manager views.
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    @property
+    def authentication_classes(self):  # pragma: no cover
+        """
+        Allow users authenticated via OAuth2 or normal session authentication.
+        """
+        from openedx.core.lib.api import authentication  # pylint: disable=import-error
+        return [
+            authentication.OAuth2AuthenticationAllowInactiveUser,
+            authentication.SessionAuthenticationAllowInactiveUser,
+        ]
+
+
+class ManagerListView(ManagerViewMixin, ListAPIView):
+    """
+        **Use Case**
+
+            * Get a list of all users that are managers for other users
+
+        **Example Request**
+
+            GET /api/user_manager/v1/managers/
+
+        **GET Parameters**
+
+            None
+
+        **GET Response Values**
+
+            If the request for information about the managers is successful, an HTTP 200 "OK"
+            response is returned with a collection of managers.
+
+            The HTTP 200 response has the following values.
+
+            * count: The number of managers in a course.
+
+            * next: The URI to the next page of results.
+
+            * previous: The URI to the previous page of results.
+
+            * num_pages: The number of pages.
+
+            * results: a list of manager users:
+
+                * id: The user id for a manager user, or null if manager doesn't have an
+                    account yet.
+
+                * email: Email address of manager.
+
+        **Example GET Response**
+
+            {
+                "count": 99,
+                "next": "https://courses.org/api/user_manager/v1/managers/?page=2",
+                "previous": null,
+                "results": {
+                    {
+                        "email": "staff@example.com",
+                        "id": 9
+                    },
+                    { ... }
+                }
+            }
+    """
+
+    serializer_class = ManagerListSerializer
+    queryset = UserManagerRole.objects.values(
+        'manager_user',
+        'manager_user__email',
+        'unregistered_manager_email',
+    ).distinct()
+
+
+class ManagerReportsListView(ManagerViewMixin, ListCreateAPIView):
+    """
+        **Use Case**
+
+            * Get a list of all users that are reports for the provided manager.
+
+            * Add a user as a report under a manger.
+
+            * Remove a user or all users under a manager.
```

Comment:
Nit: Maybe it's somewhat obvious, but this doesn't clearly say which of the following GET/POST/DELETE/DELETE forms are tied to which of these actions. Could we state each use case + example + parameters + response one at a time, instead of all the use cases, then all the examples, etc. ?

Reply:
Yeah, I find this format confusing as well, but it is something I copied from somewhere in edx-platform. Is it OK if we go for a completely different and tighter format?

Reply:
> Yeah, I find this format confusing as well, but it is something I copied from somewhere in edx-platform. Is it OK if we go for a completely different and tighter format?

Yeah, I don't see why not.